sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/UnnamedSessionManagerIntegrationTest.java

```diff
@@ -92,10 +95,11 @@ void singleUnnamedSession(MessagingEntityType entityType) {
                 .assertNext(context -> assertMessageEquals(sessionId, messageId, contents, context))
                 .assertNext(context -> assertMessageEquals(sessionId, messageId, contents, context))
                 .assertNext(context -> assertMessageEquals(sessionId, messageId, contents, context))
-                .expectComplete()
+                .thenCancel()
                 .verify(Duration.ofMinutes(2));
         } finally {
             subscription.dispose();
+            completeMessages(receiver, lockTokens, sessionId);
```

Comment:
This method is only used once. Why don't we roll in the functionality.

Reply:
I think, it is cleaner and improve readability.

Reply:
I don't think so. Methods are reusable pieces of code.

Reply:
This is an extra 3 lines. or one statement.